app/routes.py

```diff
@@ -0,0 +1,218 @@
+from app import app, db
+from flask import render_template, flash, redirect, request, url_for, session, make_response
+from app.forms import RegisterForm, LoginForm, AddTaskForm
+from flask_login import current_user, login_user, login_required
+from app.models import Users, TaskList
+from datetime import datetime
+
+@app.route('/')
+@app.route('/home')
+def home():
+	user = {'username': 'rushi'}
+	return render_template('home.html')
+
+# RegisterForm
+@app.route('/register', methods=['GET'])
+def register_get():
+	form = RegisterForm()
+	return render_template('register.html', form = form)
+
+
+@app.route('/register', methods=['POST'])
+def register_post():
+	form = RegisterForm(request.form)
+
+	# Get the values from the form
+	if form.validate():
+		# TODO: use managers and don't put logic in views
+		name = form.name.data 
+		email = form.email.data
+		username = form.username.data
+		password = form.password.data
+
+		user = Users(name=name, email=email, username=username)
+		user.set_password(password)
+		db.session.add(user)
+		db.session.commit()
+
+		flash("You are now registered and can login", "success")
+
+		return redirect(url_for('login_get'))
+
+	flash("Check the values", "danger")
+	return render_template("register.html", form=form)
+
+# LoginForm
+@app.route('/login', methods=['GET'])
+def login_get():
+	if current_user.is_authenticated:
+		return redirect(url_for('userhome'))
+	form = LoginForm()
+	return render_template('login.html', form=form)
+
+@app.route('/login', methods=['POST'])
+def login_post():
+	if current_user.is_authenticated:
+		return redirect(url_for('userhome'))
+
+	form = LoginForm(request.form)
+	if form.validate():
+		# Get form fields
+		username = form.username.data
+		password = form.password.data
+		
+		# Get details from database
+		user = Users.query.filter_by(username=username).first()
+		
+		# Check for username and password
+		if user is None or not user.check_password(password):
+			flash("User doesnt exist or password is incorrect")
+			return redirect(url_for('login_get'))
+		else:
+			login_user(user)								# Authentication of User
+			session['user_id'] = user.id
+			return redirect('userhome')
+		
+
+# User Home Page
+@app.route('/userhome')
+@login_required
+def userhome():
+	# Get the tasks to homepage
+	# Retreive data
+	tasks = current_user.tasks.all()
+	print(tasks)
+	
+	if len(tasks) > 0 :
+		# return render_template('userhome.html', tasks = tasks)
+		response = make_response(render_template('userhome.html', tasks = tasks))
+		response.headers['Last-Modified'] = datetime.now()
+		response.headers['Cache-Control'] = 'no-store, no-cache, must-revalidate, post-check=0, pre-check=0, max-age=0'
+		response.headers['Pragma'] = 'no-cache'
+		response.headers['Expires'] = '-1'
+		return response
+	else:
```

Comment:
Why is this `else` needed? You are doing the exact same thing as above. Handle it in same code.

Reply:
done